src/cp/cp.rs

```diff
@@ -827,6 +828,48 @@ fn copy_file(source: &Path, dest: &Path, options: &Options) -> CopyResult<()> {
         CopyMode::Copy    => { fs::copy(source, dest).context(&*context_for(source, dest))?; },
         CopyMode::SymLink => { symlink_file(source, dest, &*context_for(source, dest))?; },
         CopyMode::Sparse  => return Err(Error::NotImplemented(OPT_SPARSE.to_string())),
+        CopyMode::Update => {
+            if dest.exists() {
+                let src_metadata = match fs::metadata(source.clone()) {
```

Comment:
Please change this to `let src_metadata = fs::metadata(source.clone())?;` (also please do the same for the statements below).